components/BottomNavigation/tests/snapshot/MDCBottomNavigationLargeItemDialogViewSnapshotTests.m

```diff
@@ -0,0 +1,215 @@
+  UIColor *backgroundColor = UIColor.redColor;
+
+  // When/Then
+  [self verifyDialogWithItem:item size:size backgroundColor:backgroundColor];
+}
+
+- (void)testGreenBackgroundColor {
+  // Given
+  UITabBarItem *item = [[self class] tabBarItemWithTitleAndImage];
+  CGSize size = [[self class] mediumDialogSize];
+  UIColor *backgroundColor = UIColor.greenColor;
+
+  // When/Then
+  [self verifyDialogWithItem:item size:size backgroundColor:backgroundColor];
+}
+
+- (void)testBlueBackgroundColor {
+  // Given
+  UITabBarItem *item = [[self class] tabBarItemWithTitleAndImage];
+  CGSize size = [[self class] mediumDialogSize];
+  UIColor *backgroundColor = UIColor.blueColor;
+
+  // When/Then
+  [self verifyDialogWithItem:item size:size backgroundColor:backgroundColor];
+}
+
+- (void)testSmallSize {
+  // Given
+  UITabBarItem *item = [[self class] tabBarItemWithTitleAndImage];
+  CGSize size = [[self class] smallDialogSize];
+  UIColor *backgroundColor = UIColor.whiteColor;
+
+  // When/Then
+  [self verifyDialogWithItem:item size:size backgroundColor:backgroundColor];
+}
+
+- (void)testLargeSize {
+  // Given
+  UITabBarItem *item = [[self class] tabBarItemWithTitleAndImage];
+  CGSize size = [[self class] largeDialogSize];
+  UIColor *backgroundColor = UIColor.whiteColor;
+
+  // When/Then
+  [self verifyDialogWithItem:item size:size backgroundColor:backgroundColor];
+}
+
+- (void)testOnlyTitle {
+  // Given
+  UITabBarItem *item = [[self class] tabBarItemWithOnlyTitle];
+  CGSize size = [[self class] mediumDialogSize];
+  UIColor *backgroundColor = UIColor.whiteColor;
+
+  // When/Then
+  [self verifyDialogWithItem:item size:size backgroundColor:backgroundColor];
+}
+
+- (void)testOnlyImage {
+  // Given
+  UITabBarItem *item = [[self class] tabBarItemWithOnlyImage];
+  CGSize size = [[self class] mediumDialogSize];
+  UIColor *backgroundColor = UIColor.whiteColor;
+
+  // When/Then
+  [self verifyDialogWithItem:item size:size backgroundColor:backgroundColor];
+}
+
+- (void)testImageAndLongTitle {
+  // Given
+  NSString *title = @"This is a really long tab bar item title";
+  UIImage *image = [UIImage mdc_testImageOfSize:CGSizeMake(kImageWidth, kImageWidth)
+                                      withStyle:MDCSnapshotTestImageStyleFramedX];
+  UITabBarItem *item = [[UITabBarItem alloc] initWithTitle:title image:image tag:0];
+  CGSize size = [[self class] mediumDialogSize];
+  UIColor *backgroundColor = UIColor.whiteColor;
+
+  // When/Then
+  [self verifyDialogWithItem:item size:size backgroundColor:backgroundColor];
+}
+
+- (void)testLongTitleOnly {
+  // Given
+  NSString *title = @"This is a really long tab bar item title";
+  UITabBarItem *item = [[UITabBarItem alloc] initWithTitle:title image:nil tag:0];
+  CGSize size = [[self class] mediumDialogSize];
+  UIColor *backgroundColor = UIColor.whiteColor;
+
+  // When/Then
+  [self verifyDialogWithItem:item size:size backgroundColor:backgroundColor];
+}
+
+#pragma mark - Helper Methods
```

Comment:
Tip: Place these helper methods at the top of the file (with `setUp` and `tearDown`) to keep the test methods together in a single block.

Reply:
Done.